cmd/server-main.go

```diff
@@ -633,22 +633,41 @@ func serverMain(ctx *cli.Context) {
 		logger.Info(color.RedBold(msg))
 	}
 
-	bootstrapTrace("initializing the server")
-	if err = initServer(GlobalContext, newObject); err != nil {
-		var cerr config.Err
-		// For any config error, we don't need to drop into safe-mode
-		// instead its a user error and should be fixed by user.
-		if errors.As(err, &cerr) {
-			logger.FatalIf(err, "Unable to initialize the server")
+	var buckets []BucketInfo
+	func() {
+		// Freeze the services until the bucket notification subsystem gets initialized.
+		freezeServices()
+		defer unfreezeServices()
```

Comment:
One change we can do here is that for some reason `initServer` or any other Init that fails or for some reason blocks - we shouldn't block S3 API fully forever.

Have some kind of a timeout on this? - i.e allow a provision to restart the pod/service with some hidden ENV that ignores freezing calls?